src/main/java/org/openstreetmap/atlas/checks/validation/intersections/SelfIntersectingPolylineCheck.java

```diff
@@ -61,27 +83,41 @@ public SelfIntersectingPolylineCheck(final Configuration configuration)
     @Override
     public boolean validCheckForObject(final AtlasObject object)
     {
-        return object instanceof Edge && ((Edge) object).isMasterEdge() || object instanceof Area
-                || object instanceof Line;
+        // Master edges excluding ineligible highway tags
+        return ((object instanceof Edge && ((Edge) object).isMasterEdge()
+                && !INELIGIBLE_HIGHWAY_TAGS.test(object))
+                // Areas
+                || (object instanceof Area)
+                // Lines excluding ineligible highway tags
+                || (object instanceof Line && !INELIGIBLE_HIGHWAY_TAGS.test(object)))
+                // Exclude waterway tags
+                && !Validators.hasValuesFor(object, WaterwayTag.class);
```

Comment:
What is the reasoning behind excluding waterways?

Reply:
- OSM has no guidelines on handling Waterway self intersections
- Waterways have unique modeling techniques that warrant it's own inspection of geometry/tag combinations
- Waterways are not used for navigation

Unless there's an issue with Atlas handling self intersecting Waterway geometries, I'd suggest keeping them out of this check